e2e-tests/tfm/cypress/e2e/journeys/case/deal-cancellation-enabled/deal-cancellation-button.spec.js

```diff
@@ -0,0 +1,212 @@
+        cy.login(PIM_USER_1);
+        cy.visit(relative(`/case/${ainDealId}/deal`));
+      });
+
+      it('should display the deal cancellation button', () => {
+        pages.caseDealPage.cancelButton().should('exist');
+      });
+    });
+
+    describe('when logged in as a non-PIM user', () => {
+      beforeEach(() => {
+        cy.login(T1_USER_1);
+        cy.visit(relative(`/case/${ainDealId}/deal`));
+      });
+
+      it('should not display the deal cancellation button', () => {
+        pages.caseDealPage.cancelButton().should('not.exist');
+      });
+    });
+  });
+
+  describe('when visiting a GEF MIN deal summary page', () => {
+    let minDealId;
+
+    before(() => {
+      cy.insertOneGefDeal(MOCK_APPLICATION_MIN, BANK1_MAKER1).then((insertedDeal) => {
+        minDealId = insertedDeal._id;
+
+        cy.updateGefDeal(minDealId, MOCK_APPLICATION_MIN, BANK1_MAKER1);
+
+        const { dealType } = MOCK_APPLICATION_MIN;
+
+        cy.submitDeal(minDealId, dealType, T1_USER_1);
+      });
+    });
+
+    after(() => {
+      cy.deleteDeals(minDealId, ADMIN);
+    });
+
+    describe('when logged in as a PIM user', () => {
+      beforeEach(() => {
+        cy.login(PIM_USER_1);
+        cy.visit(relative(`/case/${minDealId}/deal`));
+      });
+
+      it('should display the deal cancellation button', () => {
+        pages.caseDealPage.cancelButton().should('exist');
+      });
+    });
+
+    describe('when logged in as a non-PIM user', () => {
+      beforeEach(() => {
+        cy.login(T1_USER_1);
+        cy.visit(relative(`/case/${minDealId}/deal`));
+      });
+
+      it('should not display the deal cancellation button', () => {
+        pages.caseDealPage.cancelButton().should('not.exist');
+      });
+    });
+  });
+
+  describe('when visiting a GEF MIA deal summary page', () => {
+    let miaDealId;
+
+    before(() => {
+      cy.insertOneGefDeal(MOCK_APPLICATION_MIA, BANK1_MAKER1).then((insertedDeal) => {
+        miaDealId = insertedDeal._id;
+
+        cy.updateGefDeal(miaDealId, MOCK_APPLICATION_MIA, BANK1_MAKER1);
+
+        const { dealType } = MOCK_APPLICATION_MIA;
+
+        cy.submitDeal(miaDealId, dealType, T1_USER_1);
+      });
+    });
+
+    after(() => {
+      cy.deleteDeals(miaDealId, ADMIN);
+    });
+
+    describe('when logged in as a PIM user', () => {
+      beforeEach(() => {
+        cy.login(PIM_USER_1);
+        cy.visit(relative(`/case/${miaDealId}/deal`));
+      });
+
+      it('should not display the deal cancellation button', () => {
+        pages.caseDealPage.cancelButton().should('not.exist');
+      });
+    });
+
+    describe('when logged in as a non-PIM user', () => {
+      beforeEach(() => {
+        cy.login(T1_USER_1);
+        cy.visit(relative(`/case/${miaDealId}/deal`));
+      });
+
+      it('should not display the deal cancellation button', () => {
+        pages.caseDealPage.cancelButton().should('not.exist');
+      });
+    });
+  });
+});
```

Comment:
Can we add tests for clicking the button? I.e should go to X

Not sure if this is setup yet, ignore if not